packages/agents/src/functions/SearchInPages.ts

```diff
@@ -0,0 +1,261 @@
+import {
+  Agent,
+  AgentFunctionResult,
+  AgentOutputType,
+  AgentVariables,
+  ChatLogs,
+  ChatMessageBuilder,
+  Chunker,
+  LlmApi,
+  Tokenizer,
+  trimText,
+} from "@evo-ninja/agent-utils";
+import axios from "axios";
+import { v4 as uuid } from "forked-agent-protocol";
+import { AgentFunctionBase } from "../AgentFunctionBase";
+import { FUNCTION_CALL_FAILED, FUNCTION_CALL_SUCCESS_CONTENT } from "../agents/Scripter/utils";
+import { AgentBaseContext } from "../AgentBase";
+import { Connection, EmbeddingFunction } from "vectordb";
+
+interface SearchInPagesFuncParameters {
+  query: string;
+  urls: string[];
+}
+
+export class SearchInPagesFunction extends AgentFunctionBase<SearchInPagesFuncParameters> {
+  constructor(
+    private chunker: Chunker,
+    private tokenizer: Tokenizer,
+    private llm: LlmApi,
+    private lanceDb: {
+      connect: () => Promise<Connection>,
+      embeddingFunction: (columnName: string) => EmbeddingFunction<string>
+    }
+  ) {
+    super();
+  }
+
+  get name(): string {
+    return "search_in_pages";
+  }
+  get description(): string {
+    return `Searches across different specific web page URLs for a given query.`;
+  }
+  get parameters() {
+    return {
+      type: "object",
+      properties: {
+        query: {
+          type: "string",
+          description: "Query to search the pages for",
+        },
+        urls: {
+          type: "array",
+          items: {
+            type: "string",
+          },
+          description: "URLs of the pages to search for the query",
+        },
+      },
+      required: ["query", "urls"],
+      additionalProperties: false,
+    };
+  }
+
+  buildExecutor(
+    _: Agent<unknown>,
+    context: AgentBaseContext
+  ): (
+    params: SearchInPagesFuncParameters,
+    rawParams?: string
+  ) => Promise<AgentFunctionResult> {
+    return async (
+      params: SearchInPagesFuncParameters,
+      rawParams?: string
+    ): Promise<AgentFunctionResult> => {
+      try {
+        const connection = await this.lanceDb.connect()
+        const table = await connection.createTable({
+          name: uuid(),
+          embeddingFunction: this.lanceDb.embeddingFunction("text"),
+          data: [{
+            text: " ",
+            id: uuid()
+          }]
+        })
+
+        for await (const url of params.urls) {
+          try {
+            const response = await this.processWebpage(url);
+            const data = response.map((chunk) => ({
+              text: chunk,
+              id: uuid(),
+            }))
+            await table.add(data)
+          } catch(e) {
+            console.log(`Failed to fetch ${url}`)
+          }
+        }
+
+        const results = await table
+          .search(params.query)
+          .limit(25)
```

Comment:
Can we get this as a const at the top of the file?
Same for some other constants in this file.